src/main/java/org/candlepin/model/Environment.java

```diff
@@ -114,32 +115,62 @@ public String getOwnerId() {
         return (owner == null) ? null : owner.getId();
     }
 
-    public void setOwner(Owner owner) {
+    public Environment setOwner(Owner owner) {
         this.owner = owner;
+        return this;
     }
 
     public Set<EnvironmentContent> getEnvironmentContent() {
         return environmentContent;
     }
 
-    public void setEnvironmentContent(Set<EnvironmentContent> environmentContent) {
+    public Environment setEnvironmentContent(Set<EnvironmentContent> environmentContent) {
         this.environmentContent = environmentContent;
+        return this;
+    }
+
+    public Environment addEnvironmentContent(EnvironmentContent envcontent) {
+        if (envcontent == null) {
+            throw new IllegalArgumentException("envcontent is null");
+        }
+
+        // Ensure it's set to this environment. Really this should be immutable and created internally.
+        envcontent.setEnvironment(this);
```

Comment:
Do we need to include this to the `setEnvironmentContent(Set<EnvironmentContent> environmentContent)` method for the elements in that collection?

Reply:
Yeah, we probably should. That whole method should probably be written to encapsulate the collection rather than just using whatever external collection it's given as well.